src/averageAges.js

```diff
@@ -37,7 +40,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = people
+    .filter(persons => persons.sex === 'f' && (people
+      .some(child => child.mother === persons.name) || !withChildren));
+
+  const calculateLifeYears = filteredWoman.map(woman =>
+    woman.died - woman.born);
```

Comment:
Here you can use destructuring.
```suggestion
  const calculateLifeYears = filteredWoman.map(({ died, born })=>
    died - born);
```